var/spack/repos/builtin/packages/raja/package.py

```diff
@@ -30,9 +30,12 @@ class Raja(CMakePackage, CudaPackage):
 
     variant('openmp', default=True, description='Build OpenMP backend')
     variant('shared', default=True, description='Build Shared Libs')
+    variant('chai', default=False, description='Build with CHAI support')
 
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', when='+cuda', type='build')
+    depends_on('chai', when='+chai')
+    depends_on('chai+cuda', when='+chai+cuda')
```

Comment:
This is my spack ignorance talking, but how will chai+cuda properly carry the +cuda_arch flag that needs to be specified?

Reply:
This is based on https://spack.readthedocs.io/en/latest/build_systems/cudapackage.html#usage

Apparently cuda_arch is automatically detected by CudaPackage.

Reply:
`cuda_arch` is not (yet) automatically detected, it needs to be set by users. I recommend setting it once in your `packages.yaml` so it applies to all packages.